File: scripts/day3/day3_scripts_Python/Day3_Application5_EPA.py
import os
import pandas as pd

# =============================================================================
# Instats Seminar: Web Scraping and Data Collection
# Day 3 - From shared workflow to participants' own data
# Application 5: Sustainability & Environment - EPA / World Bank
# =============================================================================
# Goal: Explore and analyze the cleaned dataset.

def explore_epa_data():
    """
    Reads the cleaned dataset from Day 2 and performs preliminary exploratory analysis.
    """
    clean_cache_file = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(__file__))), 
                                    'day2', 'data', 'cache', 'day2_app5_co2_clean.csv')
    
    if not os.path.exists(clean_cache_file):
        print(f"Error: Cleaned data file not found at {clean_cache_file}")
        return

    print(f"Loading cleaned data from {clean_cache_file}")
    df = pd.read_csv(clean_cache_file)
    
    print("\n--- Exploratory Analysis: Environmental / Population Data ---")
    print(f"Total records: {len(df)}")
    
    if 'Population' in df.columns:
        print("\nSummary Statistics for Population:")
        print(df['Population'].describe())
        
        print("\nTop 5 Regions/Countries by Population in 2020:")
        top_5 = df.sort_values(by='Population', ascending=False).head(5)
        print(top_5[['CountryName', 'Population']])
        
    print("\nMissing Values Check:")
    print(df.isnull().sum())

if __name__ == "__main__":
    explore_epa_data()

File: scripts/day3/day3_scripts_R/Day3_Application5_EPA.R
# =============================================================================
# Instats Seminar: Web Scraping and Data Collection
# Day 3 - From shared workflow to participants' own data
# Application 5: Sustainability & Environment - EPA / World Bank
# =============================================================================
# Goal: Explore and analyze the cleaned dataset.

library(dplyr)
library(fs)

explore_epa_data <- function() {
  clean_cache_file <- path(path_dir(path_dir(getwd())), "day2", "data", "cache", "day2_app5_co2_clean.csv")
  
  if (!file_exists(clean_cache_file)) {
    cat("Error: Cleaned data file not found at", clean_cache_file, "\n")
    return()
  }
  
  cat("Loading cleaned data from", clean_cache_file, "\n")
  df <- read.csv(clean_cache_file)
  
  cat("\n--- Exploratory Analysis: Environmental / Population Data ---\n")
  cat("Total records:", nrow(df), "\n")
  
  if ("Population" %in% names(df)) {
    cat("\nSummary Statistics for Population:\n")
    print(summary(df$Population))
    
    cat("\nTop 5 Regions/Countries by Population in 2020:\n")
    top_5 <- df %>% 
      arrange(desc(Population)) %>% 
      head(5) %>% 
      select(CountryName, Population)
    print(top_5)
  }
  
  cat("\nMissing Values Check:\n")
  print(colSums(is.na(df)))
}

explore_epa_data()
